infra/aws/terraform/management-account/s3.tf

```
@@ -0,0 +1,155 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+# Cost and Usage
+
+resource "aws_kms_key" "cur_bucket_key" {
+  enable_key_rotation = true
+  multi_region = true
+}
+
+resource "aws_kms_key" "cur_integrations_bucket_key" {
+  enable_key_rotation = true
+  multi_region = true
+}
+
+module "cur_reports_s3_bucket" {
+  source = "terraform-aws-modules/s3-bucket/aws"
+
+  bucket = "k8s-infra-cur-reports-bucket"
+  acl    = "private"
+
+  # S3 bucket-level Public Access Block configuration
+  block_public_acls       = true
+  block_public_policy     = true
+  ignore_public_acls      = true
+  restrict_public_buckets = true
+
+  attach_deny_insecure_transport_policy = true
+  attach_policy = true
+  policy = data.aws_iam_policy_document.cur_reports_s3_bucket.json
+
+  server_side_encryption_configuration = {
+    rule = {
+      apply_server_side_encryption_by_default = {
+        kms_master_key_id = aws_kms_key.cur_bucket_key.arn
+        sse_algorithm     = "aws:kms"
+      }
+    }
+  }
+
+  versioning = {
+    enabled    = true
+    mfa_delete = false
+  }
+}
+
+
+module "cur_reports_integrations_s3_bucket" {
+  source = "terraform-aws-modules/s3-bucket/aws"
+
+  bucket = "cur_reports_integrations_s3_bucket"
+  acl    = "private"
+
+  # S3 bucket-level Public Access Block configuration
+  block_public_acls       = true
+  block_public_policy     = true
+  ignore_public_acls      = true
+  restrict_public_buckets = true
+
+  attach_deny_insecure_transport_policy = true
+  attach_policy = true
+  policy = data.aws_iam_policy_document.cur_reports_integrations_s3_bucket.json
+
+  server_side_encryption_configuration = {
+    rule = {
+      apply_server_side_encryption_by_default = {
+        kms_master_key_id = aws_kms_key.cur_integrations_bucket_key.arn
+        sse_algorithm     = "aws:kms"
+      }
+    }
+  }
+
+  versioning = {
+    enabled    = true
+    mfa_delete = false
+  }
+}
+
+
+data "aws_iam_policy_document" "cur_reports_s3_bucket" {
+  version = "2008-10-17"
+
+  statement {
+    sid    = "AWSBillingDeliveryAclCheck"
+    effect = "Allow"
+    principals {
+      type = "Service"
+      identifiers = [
+        "billingreports.amazonaws.com"
+      ]
+    }
+    actions = [
+      "s3:GetBucketPolicy",
+      "s3:GetBucketAcl"
+    ]
```

```suggestion
    ]
    
    condition {
      test     = "StringEquals"
      variable = "aws:SourceArn"
      values   = ["arn:aws:cur:us-east-1:${data.aws_caller_identity.current.account_id}:definition/*"]
      # I think this region is ALWAYS us-east-1 but I'm not certain
    }

    condition {
      test     = "StringEquals"
      variable = "aws:SourceAccount"
      values   = [data.aws_caller_identity.current.account_id]
    }
```
along with a `data "aws_caller_identity" "current" {}` somewhere else

---

Conditions added.